File: backend/src/main/java/com/ssafy/db/entity/Pet/PetStat.java
package com.ssafy.db.entity.Pet;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;

@Getter
@Setter
@Entity
//Pet의 스탯 - type에 영향을 주는 속성
public class PetStat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pet_stat_id")
    private Long petStat_id;

    @OneToOne
    @JoinColumn(name="pet_id")
    private Pet pet;

    //TYPE 1
    @ColumnDefault(value = "0")
    private int physical;

    //TYPE 2
    @ColumnDefault(value = "0")
    private int artistic;

    //TYPE 3
    @ColumnDefault(value = "0")
    private int intelligent;

    //TYPE 4
    @ColumnDefault(value = "0")
    private int inactive;

    //TYPE 5
    @ColumnDefault(value = "0")
    private int energetic;

    //TYPE 6
    @ColumnDefault(value = "0")
    private int etc;
}

File: mobile/app/src/main/java/com/team/teamrestructuring/dto/PetStat.kt
package com.team.teamrestructuring.dto

import com.google.gson.annotations.SerializedName

data class PetStat(
    @SerializedName("petStat_id")
    var petStat_id:Long,

    @SerializedName("pet")
    var pet:Pet,

    @SerializedName("physical")
    var physical:Int,

    @SerializedName("artistic")
    var artistic:Int,

    @SerializedName("intelligent")
    var intelligent:Int,

    @SerializedName("inactive")
    var inactive:Int,

    @SerializedName("energetic")
    var energetic:Int,

    @SerializedName("etc")
    var etc:Int
){
    override fun toString(): String {
        return "PetStat(petStat_id=$petStat_id, pet=$pet, physical=$physical, artistic=$artistic, intelligent=$intelligent, inactive=$inactive, energetic=$energetic, etc=$etc)"
    }
}
